File: app/src/main/webapp/js/classes/review-vote.js
class ReviewVote {
    static fromJson(json) {
        const reviewVoteAccount = new Account();
        reviewVoteAccount.setId(json.accountId);

        const reviewVote = new ReviewVote();
        reviewVote.setId(json.id);
        reviewVote.setAccount(reviewVoteAccount);
        reviewVote.setCreatedDate(json.createdDate);
        reviewVote.setIsUpvote(json.isUpvote);

        return reviewVote;
    }

    static toJson(reviewVote) {
        const id = reviewVote.getId();
        const accountId = reviewVote.getAccount().getId();
        const createdDate = reviewVote.getCreatedDate();
        const isUpvote = reviewVote.isUpvote();

        return {
            reviewId:       id,
            accountId:      accountId,
            createdDate:    createdDate,
            isUpvote:       isUpvote
        };
    }

    constructor() {
        this._id    = null;
        this._account = null;
        this._createdDate = null;
        this._isUpvote = null;
    }

    setId(id) {
        this._id = id;
    }

    getId() {
        return this._id;
    }

    setAccount(account) {
        this._account = account;
    }

    getAccount() {
        return this._account;
    }

    setCreatedDate(createdDate) {
        this._createdDate = createdDate;
    }

    getCreatedDate() {
        return this._createdDate;
    }

    setIsUpvote(isUpvote) {
        this._isUpvote = isUpvote;
    }

    isUpvote() {
        return this._isUpvote;
    }
}

File: app/src/main/java/com/softwareverde/tidyduck/ReviewVote.java
package com.softwareverde.tidyduck;

import java.util.Date;

public class ReviewVote {
    private Long _id;
    private Long _reviewId;
    private Account _account;
    private Date _createdDate;
    private boolean _isUpvote;

    public Long getId() {
        return _id;
    }

    public void setId(final Long id) {
        _id = id;
    }

    public Long getReviewId() { return _reviewId; }

    public void setReviewId(final Long reviewId) { _reviewId = reviewId; }

    public Account getAccount() {
        return _account;
    }

    public void setAccount(final Account account) {
        _account = account;
    }

    public Date getCreatedDate() {
        return _createdDate;
    }

    public void setCreatedDate(final Date createdDate) {
        _createdDate = createdDate;
    }

    public boolean isUpvote() { return _isUpvote; }

    public void setIsUpvote(final boolean isUpvote) { _isUpvote = isUpvote; }
}
